---a/net/minecraft/entity/item/EntityExpBottle.java
+++b/net/minecraft/entity/item/EntityExpBottle.java
@@ -6,49 +6,42 @@
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
-public class EntityExpBottle extends EntityThrowable
-{
+public class EntityExpBottle extends EntityThrowable {
+
     private static final String __OBFID = "CL_00001726";
 
-    public EntityExpBottle(World worldIn)
-    {
+    public EntityExpBottle(World worldIn) {
         super(worldIn);
     }
 
-    public EntityExpBottle(World worldIn, EntityLivingBase p_i1786_2_)
-    {
+    public EntityExpBottle(World worldIn, EntityLivingBase p_i1786_2_) {
         super(worldIn, p_i1786_2_);
     }
 
-    public EntityExpBottle(World worldIn, double p_i1787_2_, double p_i1787_4_, double p_i1787_6_)
-    {
+    public EntityExpBottle(World worldIn, double p_i1787_2_, double p_i1787_4_, double p_i1787_6_) {
         super(worldIn, p_i1787_2_, p_i1787_4_, p_i1787_6_);
     }
 
-    protected float getGravityVelocity()
-    {
+    protected float getGravityVelocity() {
         return 0.07F;
     }
 
-    protected float getVelocity()
-    {
+    protected float getVelocity() {
         return 0.7F;
     }
 
-    protected float getInaccuracy()
-    {
+    protected float getInaccuracy() {
         return -20.0F;
     }
 
-    protected void onImpact(MovingObjectPosition p_70184_1_)
-    {
-        if (!this.worldObj.isRemote)
-        {
-            this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
+    protected void onImpact(MovingObjectPosition p_70184_1_) {
+        if (!this.worldObj.isRemote) {
+            // CraftBukkit - moved to after event
+            // this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
             int i = 3 + this.worldObj.rand.nextInt(5) + this.worldObj.rand.nextInt(5);
 
-            while (i > 0)
-            {
+            // CraftBukkit start
+            org.bukkit.event.entity.ExpBottleEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callExpBottleEvent(this, i);
+            i = event.getExperience();
+            if (event.getShowEffect()) {
+                this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
+            }
+            // CraftBukkit end
+
+            while (i > 0) {
                 int j = EntityXPOrb.getXPSplit(i);
                 i -= j;
                 this.worldObj.spawnEntityInWorld(new EntityXPOrb(this.worldObj, this.posX, this.posY, this.posZ, j));
@@ -57,4 +57,4 @@
             this.setDead();
         }
     }
-}
+}
